libraries/extensions/dora-record/src/main.rs

```diff
@@ -52,46 +55,69 @@ fn main() -> eyre::Result<()> {
                             std::fs::create_dir_all(&dataflow_dir)
                                 .context("could not create dataflow_dir")?;
                         }
-                        let file = std::fs::File::create(dataflow_dir.join(format!("{id}.arrow")))
-                            .context("Couldn't create write file")?;
+                        let file =
+                            tokio::fs::File::create(dataflow_dir.join(format!("{id}.parquet")))
+                                .await
+                                .context("Couldn't create write file")?;
+                        let mut writer = AsyncArrowWriter::try_new(
+                            file,
+                            schema.clone(),
+                            0,
+                            Some(
+                                WriterProperties::builder()
+                                    .set_compression(parquet::basic::Compression::BROTLI(
+                                        BrotliLevel::default(),
+                                    ))
+                                    .build(),
+                            ),
+                        )
+                        .context("Could not create parquet writer")?;
+                        let (tx, mut rx) = mpsc::channel(10);
 
-                        let writer = StreamWriter::try_new(file, &schema).unwrap();
-                        let mut writer = writer;
-                        write_event(&mut writer, data.into(), &metadata, schema.clone())
-                            .context("could not write first record data")?;
-                        writers.insert(id.clone(), (writer, schema));
+                        // Per Input thread
+                        let join_handle = tokio::spawn(async move {
+                            while let Some((data, metadata)) = rx.recv().await {
+                                if let Err(e) =
+                                    write_event(&mut writer, data, &metadata, schema.clone()).await
+                                {
+                                    println!("Error writing event data into parquet file: {:?}", e)
+                                };
+                            }
+                            writer.close().await
+                        });
+                        writers.insert(id, (tx, join_handle));
```

Comment:
We should do a `tx.send` of the `data` before inserting `tx` into `writers`, otherwise we're skipping the first input.